### PR TITLE
fix for Deepgram settings not merging properly

### PR DESCRIPTION
**Pull request by kwindla**
This code fixes the Deepgram service implementation so that this code works.

```
        stt = DeepgramSTTService(
            api_key=os.getenv("DEEPGRAM_API_KEY"),
            live_options=LiveOptions(language=Language.FR),
        )
```

Previously, this code resulted on transcription not working, because the Deepgram settings were overwritten with only the provided LiveOption values. Critical settings like `sample_rate` were overwritten as `None` rather than using appropriate default values for Pipecat.

### CODE REVIEW

**Review by markbackman — APPROVED** [review]
LGTM

File: src/pipecat/services/deepgram.py [hunk 1]
```diff
@@ -129,15 +134,19 @@ def __init__(
...
-        **kwargs,
-    ):
-        super().__init__(**kwargs)
+        )
 
-        self._settings = vars(live_options)
+        merged_options = default_options
+        if live_options:
+            merged_options = LiveOptions(**{**default_options.to_dict(), **live_options.to_dict()})
+        self._settings = merged_options
```
Comment (markbackman) [inline src/pipecat/services/deepgram.py:142]: In running this locally, I see that _update_settings expects _settings to be a Dict.
```suggestion
        self._settings = merged_options.to_dict()
```

**Comment by markbackman** [comment]
I approved prematurely. In testing other things with this, I noticed I was getting an error. Other than what I proposed above, this should do the trick.

File: src/pipecat/services/deepgram.py [hunk 2]
```diff
@@ -118,7 +119,11 @@ def __init__(
         *,
         api_key: str,
         url: str = "",
-        live_options: LiveOptions = LiveOptions(
+        live_options: LiveOptions = None,
```
Comment (aconchillo) [inline src/pipecat/services/deepgram.py:122]: Should this be `Optional[LiveOptions]`?

**Comment by aconchillo** [comment]
Good catch! In other service we have `InputParams()` which has default values and those are used properly. IN here we use `LiveOptions` directly (which doesn't have default) so we were losing all our defaults.

**Pull request merged**